### PR TITLE
New demo: Toxic comment warning

### PR DESCRIPTION
**Pull request by maudnals**
Client-side inference with Transformers.js.
See all details in https://github.com/GoogleChromeLabs/web-ai-demos/blob/a5a8302c1c265f75148c6e2fe4f52f56dd2fa5e9/toxic-review-warning/README.md
Behavior:
GIF-toxic-standalone

### CODE REVIEW

File: toxic-review-warning/src/script.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+function updateToxicityVisibility(isVisible) {
+  isToxicityVisible = isVisible;
+  document.getElementById('aiWrapper').className = isToxicityVisible
+    ? 'visible'
+    : 'hidden';
+}
+
+function handleUserInputChange(event) {
+  if (isToxicityVisible === true) {
```
Comment (tomayac) [inline toxic-review-warning/src/script.js:25]:
```suggestion
  if (isToxicityVisible) {
```

File: toxic-review-warning/src/script.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+let typingTimeout = 0;
+
+// DOM manipulation, display and events
+
+function updateToxicityVisibility(isVisible) {
+  isToxicityVisible = isVisible;
+  document.getElementById('aiWrapper').className = isToxicityVisible
+    ? 'visible'
+    : 'hidden';
+}
```
Comment (tomayac) [inline toxic-review-warning/src/script.js:22]: Can this use just one class  like `visible` that you then [`toggle()`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle), or, maybe better, just the `hidden` attribute, since this is what you want in the end, the thing to be shown or not.

File: toxic-review-warning/src/script.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    clearTimeout(typingTimeout);
+  }
+  typingTimeout = setTimeout(() => {
+    console.info('Input: User has stopped typing');
+    if (modelStatus === MODEL_STATUS.READY) {
+      console.info('Trigger inference');
+      runLLMInference();
+    }
+    setTimeout(() => {}, 1000);
+  }, TYPING_DELAY); // 800ms delay
```
Comment (tomayac) [inline toxic-review-warning/src/script.js:38]:
```suggestion
  }, TYPING_DELAY);
```

If you want to comment on a delay that works well in practice, do it where the delay is set.

File: toxic-review-warning/src/script.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  if (typingTimeout) {
+    clearTimeout(typingTimeout);
+  }
+  typingTimeout = setTimeout(() => {
+    console.info('Input: User has stopped typing');
+    if (modelStatus === MODEL_STATUS.READY) {
+      console.info('Trigger inference');
+      runLLMInference();
+    }
+    setTimeout(() => {}, 1000);
```
Comment (tomayac) [inline toxic-review-warning/src/script.js:37]: Why is this needed? Is the idea to show if inference is currently running? Then probably better to add a flag in the form of a global variable.

File: toxic-review-warning/src/script.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  document.getElementById('reviewInputEl').value = '';
+  updateToxicityVisibility(false);
+  window.alert('Review posted!');
+}
+
+// Gen AI / Inference + Worker message handling
+
+function runLLMInference() {
+  // TODO manage empty input
+  const userPrompt = document.getElementById('reviewInputEl').value.trim();
```
Comment (tomayac) [inline toxic-review-warning/src/script.js:72]:
```suggestion
  const userPrompt = document.getElementById('reviewInputEl').value.trim();
  if (!userPrompt) {
    return;
  }
```

File: toxic-review-warning/src/index.html [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                <div class="ai-effect-wrapper">
+                  <div id="aiEffect"></div>
+                </div>
+                <div id="toxicityAssessmentEl"></div>
+              </div>
+              <button onclick="simulatePostReview()">
+                Post
+              </button>
+            </div>
+          </div>
```
Comment (tomayac) [inline toxic-review-warning/src/index.html:44]: Wrap this in a `<form>` and make the `<button type=submit>`. Remember to listen for `FormElement.onsubmit` and cancel the default behavior.

Comment (maudnals) [reply toxic-review-warning/src/index.html]: Done 👍

File: toxic-review-warning/src/script.js (hunk 4 above)
Comment (maudnals) [reply toxic-review-warning/src/script.js]: Obsolete code, used to have something in the function. Removed - thanks.

File: toxic-review-warning/src/script.js (hunk 2 above)
Comment (maudnals) [reply toxic-review-warning/src/script.js]: A simple `toggle` wouldn't do it, since this visibility depends on multiple factors - but TIL `toggle` takes a second argument (`...`, boolean) which is convenient and does simplify the code a bit 👍 Done.

(Note: I like classes a bit better than `hidden` in general, since any `display:` CSS property overrides the `hidden` attribute, and I feel like I end up needing to add a new wrapper and tweak my layout.)

**Review by andreban — APPROVED** [review]
LGTM

**Review by tomayac — APPROVED** [review]
LGTM 2

Thanks for addressing the previous comments!
